packages/react/src/primitives/Link/Link.tsx

```diff
@@ -7,19 +7,19 @@ import { View } from '../View';
 import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
 const LinkPrimitive: Primitive<LinkProps, 'a'> = (
-  { as = 'a', children, className, isExternal, to, ...rest },
+  { as, children, className, isExternal, to, ...rest },
   ref
 ) => {
   useDeprecationWarning({
-    shouldWarn: to != null,
+    shouldWarn: to != null && !as,
     message:
       "The Link component's to prop will soon be deprecated. " +
       'Please see the Amplify UI documentation for using the Link component with routing libraries: ' +
       'https://ui.docs.amplify.aws/react/components/link#routing-libraries',
   });
   return (
     <View
-      as={as}
+      as={as ?? 'a'}
```

Comment:
Should we use a ternary here to be more explicit?

Reply:
+1

Reply:
Like this?

`as ? as : 'a'`